BASE/src/Microsoft.ApplicationInsights/Extensibility/TelemetryConfiguration.cs

```diff
@@ -64,6 +69,8 @@ static TelemetryConfiguration()
                     Activity.ForceDefaultIdFormat = true;
                 }                
             });
+
+            SelfDiagnostics = SelfDiagnosticsProvider.EvaluateSelfDiagnosticsConfig<SelfDiagnosticsFileWriter>();
```

Comment:
One issue with making this inside Static ctor is - there is no way to dispose.

Reply:
probably best to move this inside the constructor.

Reply:
and dispose the file stuff when we dispose everything else.

Reply:
StreamWriter is created in a using and is immediately disposed.
But the EventListener that subscribes to ETW events does need to be disposed :(

I could move this to the normal ctor, but we would still only want one instance of the file logger.
This creates a new problem.. how do I properly dispose a single instance of the filelogger if 2+ TelemetryConfigurations have access to it??

Reply:
in the static scenario, we can subscribe to the Process Exit and dispose then. Would look like this: https://stackoverflow.com/a/13258842/1466768